devel/validate-platforms

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+#
+# Validate different platform builds of the final Nextstrain image.
+#
+set -euo pipefail
+
+# Set default values.
+registry=localhost:5000
+tag=latest
+
+# Read command-line arguments.
+while getopts "r:t:" opt; do
+    case "$opt" in
+        r) registry="$OPTARG";;
+        t) tag="$OPTARG";;
+        *) echo "Usage: $0 [-r <registry>] [-t <tag>]" 1>&2; exit 1;;
+    esac
+done
+
+IMAGE="$registry/nextstrain/base:$tag"
+PLATFORMS=(linux/amd64 linux/arm64)
+
+main() {
+    # Check software where downloaded versions are not deterministic at
+    # different points in time (e.g. if "latest" is downloaded).
+
+    local report_dir
+    report_dir="$(mktemp -dt "$(basename "$0")"-XXXXXX)"
+
+    for platform in "${PLATFORMS[@]}"; do
+        echo "[$platform] Pulling image..."
+        docker pull -q --platform "$platform" "$IMAGE"
+
+        echo "[$platform] Checking that the platform is expected..."
+        check-platform "$platform"
+
+        # Initialize a directory for the report file, ensuring slashes in the
+        # platform name are subdirs.
+        report="$report_dir/$platform"
+        echo "[$platform] Generating report file: $report"
+        mkdir -p "$(dirname "$report")"
+
+        # Create a report file for the platform.
+        echo "[$platform] Determining software versions..."
+        docker-run "$platform" bash -c '
+            PS4="\$ "
+            set -x
+
+            nextstrain --version
+            nextalign --version
+            nextclade --version
+            augur --version
+            auspice --version
```

Comment:
```suggestion
            auspice --version
            treetime --version
```

We definitely need to add `treetime` here - it's at the centre of much of `augur`.

The reason it may not be in this list is because it was created based on the way we install tools (e.g. binary vs pip or pypi) but that's not how it should be. But at least from my perspective what matters is what's in there, not how it ended up there.

Reply:
The current list is 1:1 with the tools under the Dockerfile section "Add unpinned programs and Nextstrain components" (see https://github.com/nextstrain/docker-base/pull/130#discussion_r1127163040).

I don't disagree with adding `treetime`, but want to be clear on moving away from the current criteria here and defining a new one.

Maybe the criteria can be updated to something like: include all tools under that section, plus any others that are worth tracking (slightly subjective, but better to err on the side of caution)? The "others" would include `treetime` and the extras you've proposed in #132.

Reply:
The current section title is misleading – I'm updating it in 6da76965daa9c74e7b0defed0fca52140e2d6c13. Doesn't matter if it's "Nextstrain" or not, the original criterion here is to keep 1:1 with unpinned tools.

[Like you said](https://github.com/nextstrain/docker-base/pull/130#issuecomment-1462167413), I think this can be merged as-is (pending proper review). More checks can be added later.

Reply:
Since there's no harm in adding a pinned tool let's add tree time as we may well unpin.

And is tree time really pinned? Or does it have minimum version.

Tests pass, so for me review passes, it can always be refactored later, this is tooling and it's not a huge issue if it churns a bit.

Reply:
I have no issue adding `treetime --version` to the list (with a comment).

Reply:
Added treetime in [force-push](https://github.com/nextstrain/docker-base/compare/c70aa6c2d37aed5572ae20f1fdfbd0c287559bb7..7d8e9480ba91af55f0a225718b405f63570a465e).

Note that `treetime` is not installed as a command (maybe that should change), so it is checked by:

```sh
python3 -c "from importlib.metadata import version; print(version(\"phylo-treetime\"))"
```

Reply:
Thanks! Do I understand correctly that it's not available as command because it's not installed explicitly, only as dependency through augur?
Don't see why we shouldn't install treetime explicitly - it's very useful to have.

Reply:
> Do I understand correctly that it's not available as command because it's not installed explicitly, only as dependency through augur?

Yes, see #139